### PR TITLE
Fixes #39282 - Show Applicable/Installable toggle for multi-CVEnv hosts

### PR DESCRIPTION
**Pull request by nofaralfasi**
The Applicable/Installable errata toggle was incorrectly hidden for
hosts assigned to multiple content view environments when the first
CVEnv contained a rolling content view, even if subsequent CVEnvs
were non-rolling.

Added new method content_view_environments_all_default_or_rolling?
to ContentFacet that returns true when:
- First CVEnv is Library (Default Org View + Library environment), OR
- All CVEnvs are rolling CVs or Library hosts

This is exposed via the API as content_view_environments_all_default_or_rolling
and consumed by the frontend to determine toggle visibility.

#### What are the changes introduced in this pull request?
This PR fixes a bug where the Applicable/Installable errata toggle was incorrectly hidden for hosts assigned to multiple content view environments (CVEnvs).

Previously, the toggle visibility logic only checked `contentFacet.contentView.rolling`, which represents only the first CVEnv. If the first CVEnv was rolling but subsequent CVEnvs were non-rolling, the toggle would be hidden even though applicable-but-not-installable errata could exist in the non-rolling CVEnvs.

The fix updates the logic to evaluate all assigned CVEnvs:
Hide toggle when first CVEnv is Library, OR all CVEnvs are rolling/Library
Show toggle when at least one non-rolling, non-Library CVEnv exists

#### Considerations taken when implementing this change?
1. Backwards compatibility: The new logic falls back to the original single-CVEnv behavior when contentViewEnvironments is not present, ensuring existing deployments continue to work.

2. Library-first hosts: Per the expected behavior, if the first CVEnv is Library (Default Org View + Library environment), the host is considered a "library host" and the toggle is hidden regardless of subsequent CVEnvs.

3. Rolling + Library combinations: A host with only rolling CVs and/or Library environments will have all applicable errata be installable, so the toggle is unnecessary and hidden.

#### What are the testing steps for this pull request?
1. Run the automated tests

2. Manual testing:
- Create a host assigned to multiple CVEnvs: [rolling CV, non-rolling CV]
- Navigate to the host's Errata tab
Expected: The Applicable/Installable toggle should be visible

- Create a host assigned to [rolling CV, rolling CV]
Expected: The toggle should be hidden

- Create a host with Library as the first CVEnv: [Library, non-rolling CV]
Expected: The toggle should be hidden

- Create a host with [non-rolling CV, Library]
Expected: The toggle should be visible

## Summary by CodeRabbit

* **Bug Fixes**
  * The Applicable/Installable errata toggle now consistently shows or hides across all environment configurations (default, rolling, library, mixed, and multi-content-view cases).

* **New Features**
  * Added a consolidated environment indicator so the UI can determine toggle visibility more reliably.

* **Tests**
  * Expanded tests and fixtures to cover the new environment shapes and edge cases.

### CODE REVIEW

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab/ErrataTab.js [hunk 1]
```diff
@@ -403,11 +403,46 @@ export const ErrataTab = () => {
...
+  // - A host assigned only to rolling CVs and/or Library environments
+  // Toggle should be SHOWN when there's at least one non-rolling, non-Library CVEnv
+  const shouldShowToggle = () => {
+    const cves = contentFacet?.contentViewEnvironments;
+
+    if (!cves || cves.length === 0) {
+      // Fall back to single CVEnv logic for backwards compatibility
+      const isLibraryHost = contentFacet?.contentViewDefault === true &&
+                            contentFacet?.lifecycleEnvironmentLibrary === true;
+      const isRolling = contentFacet?.contentView?.rolling === true;
```
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/Tabs/ErrataTab/ErrataTab.js]: `contentFacet.contentView` only checks the first assigned content view environment. It should not be used here.

**Review by jeremylenz — CHANGES_REQUESTED** [review]

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab/ErrataTab.js (hunk 1 above)
Comment (nofaralfasi) [reply webpack/components/extensions/HostDetails/Tabs/ErrataTab/ErrataTab.js]: You're right, my bad.

File: webpack/components/extensions/HostDetails/Tabs/__tests__/errataTab.test.js [hunk 2]
```diff
@@ -596,115 +597,251 @@ test('Toggle Group shows if it\'s not the default content view or library enviro
 });
 
 test('Toggle Group shows if it\'s the default content view but non-library environment', async (done) => {
```
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/Tabs/__tests__/errataTab.test.js]: Default Organization View only exists in the Library lifecycle environment, so having this test at all doesn't make sense to me.  The reverse - Library LCE, non-default CV - would make much more sense. update: I see we already have that test below. So I'd say delete this one.

File: webpack/components/extensions/HostDetails/Tabs/ErrataTab/ErrataTab.js [hunk 3]
```diff
@@ -403,11 +403,42 @@ export const ErrataTab = () => {
...
+    const allRollingOrLibrary = cves.every((cve) => {
+      const isRolling = cve.content_view?.rolling === true;
+      return isRolling || isLibraryHost(cve);
+    });
+    if (allRollingOrLibrary) {
+      return false;
+    }
+
+    // Otherwise, show toggle (at least one non-rolling, non-Library CVEnv exists)
+    return true;
```
Comment (jeremylenz) [inline webpack/components/extensions/HostDetails/Tabs/ErrataTab/ErrataTab.js]: Thinking about it some more, I would prefer all this logic to be computed on the back end.

The content_facet model has a method, `def default_environment?` that already computes this, but doesn't take into account rolling content views. So my suggestion would be

1. update the method logic to incorporate rolling CVs
2. Add a boolean field to content facet rabl, `content_view_environments_default`
3. Consume this new field here on the frontend.

Comment (nofaralfasi) [reply webpack/components/extensions/HostDetails/Tabs/ErrataTab/ErrataTab.js]: > Thinking about it some more, I would prefer all this logic to be computed on the back end.

Agreed - moving the logic to the backend is definitely the right call. 

> The content_facet model has a method, `def default_environment?` that already computes this, but doesn't take into account rolling content views. 

The existing `default_environment?` method is used in `katello_urls_helper.rb` for content path generation during provisioning. That code checks if a host is using the Default Organization View + Library environment to determine whether to include the content view label in the repository path.

If we modified `default_environment?` to also return true for rolling content views, it would break content path generation - rolling CV hosts would incorrectly skip adding their CV label to paths, causing provisioning issues.

So we created a separate method `content_view_environments_all_default_or_rolling?` specifically for determining errata toggle visibility, keeping the original `default_environment?` behavior intact.

**Comment by jeremylenz** [comment]
I love how much this is simplified now!

seems eslint needs some love

**Comment by jeremylenz** [comment]
@nofaralfasi Need to either squash or fix the second commit message.

**Review by jeremylenz — APPROVED** [review]
Thanks @nofaralfasi!

works as expected

**Comment by nofaralfasi** [comment]
> works as expected

Can you merge it? I don't have permissions.
